Use python's builtin logging instead of dirrectly printing

nitanmarcel opened the pull request:
**Checklist**

- [ ] Closing issues: #issue
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some documentation

**Description**

Makes use of python's built-in logging library to control which logs get displayed based on an environment variable.

Right now it's a base, and it need to be revised based on what where should go.

Built-in commands are:

LOGGING.debug
LOGGING.info
LOGGING.warn
LOGGING.error
LOGGING.critical

The levels can be controlled at runtime via the -V command, or by R2API_LOG= environment.

Option to also push to a file exists with R2AI_LOGFILE=. 

I'm thinking of making it push to both file and console by default and give r2ai it's own directory in .local. [appdirs](https://pypi.org/project/appdirs/) could make picking up directories easier but can be done manually too.

The logs will be structured/session with names like, r2ai.log (for last session), r2ai.log.1, r2ai.log.2 etc.

What do you think? Maybe also move everything else to it's own config file

trufae commented on r2ai/large.py:
```diff
@@ -62,8 +63,7 @@ def keywords_ai(self, text):
         response = mm(msg, stream=False, temperature=0.001, stop="</s>", max_tokens=1750)
-        if self.ai.env["debug"] == "true":
-            print("KWSPLITRESPONSE", response)
+        LOGGING.debug("KWSPLITRESPONSE", response)
```
XDD

trufae commented on r2ai/interpreter.py:
```diff
@@ -713,8 +711,7 @@ def keywords_ai(self, text):
         response = mm(msg, stream=False, temperature=0.1, stop="</s>", max_tokens=1750)
-        if self.env["debug"] == "true":
-            print("KWSPLITRESPONSE", response)
+        LOGGER.debug("KWSPLITRESPONSE", response)
```
use meaningful text here

nitanmarcel replied:
Hmm, that was there already xD

Should be network split response I guess?

nitanmarcel replied:
On probably just response?

nitanmarcel replied:
Or just logger.debug(response)

nitanmarcel commented:
I've added childers to the loggers. for example for openai requests it will show the prefix `r2ai.openai` r2ai being the base and openai the endpoint used. and I removed the KWSPLITRESPONSE completly. since it can be identified by the log tags

nitanmarcel commented:
since we already have rich and it has a nice handler to make the logging pop more with colors based on their level

![Screenshot from 2024-09-09 20-31-38](https://github.com/user-attachments/assets/ad6e3c60-5e07-48d2-8487-6473a86a2091)

trufae reviewed (changes requested):
^

nitanmarcel commented:
> ^

can you redo? for some reason they don't show up for me..
